packages/aws-cdk-lib/aws-dynamodb/lib/table-v2.ts

```diff
@@ -138,10 +138,10 @@ export interface TableOptionsV2 {
 
   /**
    * Kinesis Data Stream to capture item level changes.
-   *
+   * can supply either a Stream or a KinesisStreamInput
    * @default - no Kinesis Data Stream
    */
-  readonly kinesisStream?: IStream;
+  readonly kinesisStream?: IStream | KinesisStreamInput;
```

Comment:
This is not allowed in CDK. JSII doesn't accept union type as it would break other languages.

I was thinking more in the format of 
```
export interface KinesisStreamInput extends IStream {
  /**
   * Kinesis Stream approximate creation timestamp prescision
   *
   * @default ApproximateCreationDateTimePrecision.MICROSECOND
   */
  readonly approximateCreationDateTimePrecision?: ApproximateCreationDateTimePrecision;
};
```